core/src/main/java/org/elasticsearch/index/mapper/TokenCountFieldMapper.java

```diff
@@ -136,7 +136,7 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
         final int tokenCount;
         if (value == null) {
-            tokenCount = (Integer) fieldType().nullValue();
+            tokenCount = 0;
```

Comment:
I'm not sure returning 0 if the value is `null` here is the right thing to do. This would mean we cannot differentiate this from the token count of an empty string (""). I'm not sure what we do in other cases like this, but maybe we shouldn't add any IndexableField at all in this case. Maybe @jpountz or @jimczi have an opinion on this.